### PR TITLE
Adding xdlopsgemm and blockwisegemm i8 implementation

### PR DESCRIPTION
**Pull request by jerryyin**
I did the following in this PR:
 - Added i8 definition for xdlops gemm and blockwise gemm op
 - Amended builderUtils to take i8 into account
 - Added unit tests for both amended ops

Sending the i8 PRs in reasonable smaller chunks so that it is easier to review. I'm hoping to get feedback such as to address issues early.

Going forward I plan on to work on following areas:
 - bottom up, next PR: continue to work on i8 conversion of step4 ops including:
   - ThreadwiseCopyV2
   - ThreadwiseStore
   - ThreadwiseLoad
   - InWarpTransposeOp  
 - top down, next PR: Start from a set of tuning parameters and lower from gridwisegemm -> blockwise level ops

### CODE REVIEW

File: mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td
Comment (krzysz00) [inline mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td]: Shouldn't those VectorOfLengthAndType<[4], [I8]> be something like [4, 16], or am I misremembering the context?

Also, while you're here, could you break those constraints up onto multiple lines - they've gotten really long.

File: mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp
Comment (krzysz00) [inline mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp:35]: Why not expose this? We might want it in miopen-gen.cpp later

**Review by krzysz00 — COMMENTED** [review]
Minor nitpicks, otherwise looks solid to me - thanks for keeping up on this!

File: mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td
Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td]: Sure, will break to make it shorter.

The reason for `VectorOfLengthAndType<[4], [I8]>` definition is that for both xdlops exposed they are only able to execute the argument of `i32`, or equivalent to 4xi8.

In the lowered code, bufferA and bufferB will load temporary result and get directly feed into the xdlops instruction therefore it has to match the length of argument with xdlops.

File: mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp
Comment (jerryyin) [reply mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp:35]: Just trying to follow the defensive programming style that we can expose it when we need it. It's going to be a fairly trivial change. By then we can judge how and what to expose as public.

But right now since there's no consumer I'm just going to hide the implementation in this compilation unit only.

File: mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td
Comment (krzysz00) [reply mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td]: Noted that both the xdlops just take i32 - for some reason, I thought some of them were wider

File: mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp
Comment (krzysz00) [reply mlir/lib/Dialect/MIOpen/utility/builderUtils.cpp:35]: Makes total sense

**Review by krzysz00 — APPROVED** [review]
Approved, though if the anti-long-lines changes go in before merge it'd be great

File: mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td
Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td]: Agreed, I think you meant to say that the data preparation time we can take advantage of kpack to better take advantage of the bandwidth, which will translate to more inner loops in xdlops execution.

Comment (jerryyin) [reply mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td]: breaking lines addressed in [700a3a9](https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/505/commits/700a3a9c5b5a3522b38d16dfc9cb77cfa984018b)

**Pull request merged**
